### DISCUSSION TITLE
age vs Kryptor

### DISCUSSION THREAD

**Discussion by hakavlad** [samuel-lucas6/Kryptor#76] [General]
Filippo about Kryptor
https://news.ycombinator.com/item?id=41159236

>I can't quite make out from https://www.kryptor.co.uk/specification if it does proper signcryption, sign-then-encrypt (vulnerable to signature stripping and re-signing), or encrypt-then-sign

Maybe need clarification in the spec.

**Comment by samuel-lucas6** [comment]
I think it's [pretty clear](https://www.kryptor.co.uk/specification#file-metadata-header) that ChaCha20-Poly1305 is being used if you read the full spec. I expect Filippo didn't look too thoroughly because there's a lot of information on the page. Signatures aren't mention for anything related to encryption.

However, I will review the website at some point. I do need to discuss [insider non-repudiation](https://neilmadden.blog/2021/02/16/when-a-kem-is-not-enough/) on the limitations page.
